### PR TITLE
Added mentees onboarding docs; Improved Community and contribution docs.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by bwplotka** [comment]
We can ignore link check it is not forward compatible (:

File: docs/contributing/mentorship.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+* [Google Summer of Code](https://github.com/cncf/mentoring/tree/master/summerofcode)
+* [Red Hat Beyond](https://research.redhat.com/blog/2020/05/24/open-source-development-course-and-devops-methodology/)
+
+## For Mentees
+
+You have been selected for mentorship with Prometheus or Thanos? Congratulations! Here is some short quick information for a quick start!
+
+### Onboarding Check List
+
+* [] Please see and acknowledge our [Code of Conduct](https://thanos.io/tip/contributing/code_of_conduct.md/). A similar one is for the Prometheus project. TL;DR: Be nice and friendly!
```
Comment (yashrsharma44) [inline docs/contributing/mentorship.md:25]: Should be `- [x]` instead of `* []`

**Review by yashrsharma44 — APPROVED** [review]
Small nits otherwise LGTM!
Thanks for the shoutout :grin:

**Comment by bwplotka** [comment]
Done!

**Comment by bwplotka** [comment]
@prmsrswt @kakkoyun  does it make sense?

**Review by prmsrswt — APPROVED** [review]

**Review by kakkoyun — APPROVED** [review]

**Pull request merged**